app/src/main/java/com/github/se/polyfit/model/meal/Meal.kt

```diff
@@ -46,20 +77,31 @@ data class Meal(
         this["mealTemp"] = data.mealTemp
         this["nutritionalInformation"] =
             NutritionalInformation.serialize(data.nutritionalInformation)
+        this["ingredients"] = data.ingredients.map { Ingredient.serialize(it) }
       }
     }
 
     fun deserialize(data: Map<String, Any>): Meal? {
       return try {
-        val mealID = data["mealID"] as Int
+        val mealID = data["mealID"] as Long
         val occasion = data["occasion"]?.let { MealOccasion.valueOf(it as String) } as MealOccasion
```

Comment:
if occasion is a mandatory param how come we need `data["occasion"]?`, shouldn't it always be defined?

Reply:
Yes, you're right; that is a relic from a previous implementation.